api/types/types.go

```diff
@@ -54,13 +54,56 @@ type ImageMetadata struct {
 	LastTagTime time.Time `json:",omitempty"`
 }
 
+// ImagePlatform describes the platform which the image in the manifest runs on.
+// This type is our equivalent of ocispec.Platform, but implements fmt.Stringer
+// interface.
+type ImagePlatform struct {
+	// Architecture field specifies the CPU architecture, for example
+	// `amd64` or `ppc64`.
+	Architecture string `json:"architecture"`
+
+	// OS specifies the operating system, for example `linux` or `windows`.
+	OS string `json:"os"`
+
+	// OSVersion is an optional field specifying the operating system
+	// version, for example on Windows `10.0.14393.1066`.
+	OSVersion string `json:"os.version,omitempty"`
```

Comment:
If we are going to define our own type, I wonder if its better to de-flatten this and make an `OS` struct with `Name`, `Version`, and `Features`

Reply:
You mean something like this?

```go
type ImagePlatform struct {
	ArchInfo
	OSInfo
}

type ArchInfo struct {
	Architecture string `json:"architecture"`
	Variant      string `json:"variant,omitempty"`
}

type OSInfo struct {
	OS         string   `json:"os"`
	OSVersion  string   `json:"os.version,omitempty"`
	OSFeatures []string `json:"os.features,omitempty"`
}
```

I tried to keep our type close to the upstream type, so that we can just cast one to the other, but still being able to "enhance" our type (implementing the stringer interface for a start). Main reason for me to define our own type is to prevent making the same mistake we did on other endpoints by reusing internal types, and modify the API without making actual changes to the API.

Did you have a specific use-case in mind for having `OS` separate?

Reply:
More like

```go
type Platform struct {
    Architecture string `json:"architecture"`
    Variant string json:"variant,omitempty"`
    OS OSInfo json:"os,omitempty"`
}

type OSInfo struct {
    Name string `json:"name"`
    Version string `json:"version"`
    // etc
}
```

Without the weird name spacing on the field names in json.
The question is more about the wire format than how we represent it in code.

Reply:
Except we don't have `{"os": {"name":"foo", "version":"bar"}}` but have literally `{"os":"foo","os.version":"bar"}` :disappointed: 

(So we'd have to implement our own marshal/unmarshal code, AFAIK)

Reply:
This is data we have in docker that we are just exposing to the API as we see fit, right?

Reply:
I think it's the image's platform data being presented on the container inspect endpoint -- there's a strong argument there for presenting the platform data in the OCI standard format, IMO. :sweat_smile:

Reply:
Works for me.